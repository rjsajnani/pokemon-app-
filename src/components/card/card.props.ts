export interface CardProps {
  name: string;
  image: string;
  id: number;
}
