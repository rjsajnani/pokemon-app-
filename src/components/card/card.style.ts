import { Dimensions, StyleSheet } from 'react-native';

export const CardStyle = StyleSheet.create({
  container: {
    height: 120,
    backgroundColor: 'black',
    borderRadius: 10,
    padding: 10,
    width: Dimensions.get('window').width * 0.45,
  },
  text: { color: 'white' },
});
