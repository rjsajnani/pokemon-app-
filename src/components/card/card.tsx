import React from 'react';
import { CardProps as Props } from './card.props';
import { CardStyle as styles } from './card.style';
import { Image, View } from 'react-native';
import { Text } from '../common';

export const Card: React.FC<Props> = (props) => {
  const { name, image, id } = props;

  return (
    <View style={styles.container}>
      <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
        <Text text={name} presets="body1" style={styles.text} />
        <Text
          text={`#${id.toString().padStart(3, '0')}`}
          presets="body1"
          style={styles.text}
        />
      </View>
      <Image source={{ uri: image }} style={styles.image} />
    </View>
  );
};
