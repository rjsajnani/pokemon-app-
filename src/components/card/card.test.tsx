import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';

import { Card } from './card';

describe('<Card />', () => {
  let instance: RenderAPI;
  beforeEach(function () {
    instance = render(
      <Card
        name="Pikachu"
        id={2}
        image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png"
      />,
    );
  });

  it('renders correctly', () => {
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('able to find the pokemon rendered', async () => {
    const { findByText } = instance;

    expect(await findByText('Pikachu')).toBeTruthy();
  });
});
