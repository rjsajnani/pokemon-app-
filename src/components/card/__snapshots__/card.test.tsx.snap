// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "black",
      "borderRadius": 10,
      "height": 120,
      "padding": 10,
      "width": 337.5,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 16,
            "letterSpacing": 0.2,
            "lineHeight": 24,
          },
          {
            "color": "white",
          },
        ]
      }
    >
      Pikachu
    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 16,
            "letterSpacing": 0.2,
            "lineHeight": 24,
          },
          {
            "color": "white",
          },
        ]
      }
    >
      #002
    </Text>
  </View>
  <Image
    source={
      {
        "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png",
      }
    }
    style={
      {
        "bottom": 2,
        "height": 80,
        "position": "absolute",
        "right": 2,
        "width": 80,
      }
    }
  />
</View>
`;
